py/stackctrl.h

```diff
@@ -26,15 +26,20 @@
 #ifndef MICROPY_INCLUDED_PY_STACKCTRL_H
 #define MICROPY_INCLUDED_PY_STACKCTRL_H
 
-#include "py/mpconfig.h"
+#include "py/mpstate.h"
 
 void mp_stack_ctrl_init(void);
 void mp_stack_set_top(void *top);
 mp_uint_t mp_stack_usage(void);
 
 #if MICROPY_STACK_CHECK
 
-void mp_stack_set_limit(mp_uint_t limit);
+inline static void mp_stack_set_limit(mp_uint_t limit) {
```

Comment:
This is no longer setting the limit, instead it's setting the total available stack space.  The limit is then that number minus the margin.

So I suggest renaming this function to something like `mp_stack_set_size()`.  If we do that, it would force users to notice this change in API.  OTOH maybe it's nicer for them to provide a backwards compatible macro `#define mp_stack_set_limit mp_stack_set_size`?

Reply:
Good point... what about `mp_stack_check_init(mp_uint_t stack_size)`. Then the name still indicates the purpose, but more accurate about what it will do.

Reply:
I think `mp_stack_check_init()` is confusing because there's already `mp_stack_ctrl_init()`... with two init functions it's confusing which one to call.

The current situation is that you either do one of the following to initialise stack checking:
- call `mp_stack_ctrl_init()` to use the current stack pointer to set the top of the stack, then call `mp_stack_set_limit()` to set the limit from there
- call `mp_stack_set_top()` to set an explicit top pointer, then call `mp_stack_set_limit()`

Then you can call `mp_stack_usage()` and `mp_stack_check()`.

If we are going to rejig the API, I'd propose to have two separate init functions:
- `mp_stack_check_init_with_sp_here(size_t stack_size)`
- `mp_stack_check_init_with_top(void *top, size_t stack_size)`

(Note the `size_t` type.)

Then it's much clearer what to call, and you only need to call one function to init it.